server/controllers/commentController.js

```diff
@@ -162,4 +162,42 @@ export default class CommentController {
     if (userPosts.length <= 0) return successStat(res, 200, 'message', 'You have not made any post so far...');
     return successStat(res, 200, 'data', userPosts);
   }
+
+  /**
+   * @static
+   * @description allows a user add a comment to an already highlighted text
+   * @param {*} req Request object
+   * @param {*} res Response object
+   * @returns {Object} Object containing the user comment, author, and timestaps
+   * @memberof CommentController
+   */
+  static async commentAhighligh(req, res) {
+    const { body: { highlight: { comment, id } }, user } = req;
+    const highlightedComment = await models.Comment.findOne({ where: { highlightId: id } });
+    if (!highlightedComment) return errorStat(res, 404, 'Post not found');
+    const { highlightUser, articleId, highlightId } = highlightedComment;
+    const newComment = {
+      highlightUser,
+      body: comment,
+      articleId,
+      highlightId,
+      authorId: user.id,
+    };
+    const userComment = await models.Comment.create(newComment);
```

Comment:
Nice implementation Jude, however, I feel you needed to have added the comment to the post too as it is a child of the post. `await post.addComment(userComment)`. If this is not the case, do you have a relationship where comments can haveMany comments?

Reply:
The highlight doesn't have a different table. it still uses the comment table.